target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom

```diff
@@ -65,6 +65,10 @@ case "$FIRMWARE" in
 		caldata_extract "radiocfg" 0x1000 0x440
 		ath9k_patch_mac $(mtd_get_mac_ascii devdata "wlan24mac")
 		;;
+	ruckus,r500)
+		caldata_extract "art" 0x41000 0x440
+		ath9k_patch_fw_mac $(mtd_get_mac_binary "art" 96) 2
```

Comment:
use 0x60 instead of 96. Why do we need an increment here although 0x60 is not used for anything else?

Are there any addresses in the caldata (0x41002, 0x41006, 0x45002, 0x45006)?

Reply:
Looking more carefully, 0x76 contains the correct MAC, no need for an increment

# hexdump -s 0x76 -n 6 /dev/mtd2
0000076 f03e 9029 bc8c

Reply:
0x41002 contains the correct MAC for ath9k

Reply:
> for example on ZF 7372, this command reveals a lot
> `fw show all`

Interesting!  Do you know a way to ask uboot to boot into the secondary image, so that I can easily test (rather than corrupt the kernel header at 0xE0) ?

current primary boot image is Image1
---------------<Image1 FW header>
Magic:        RCKS
next_image:   0x130000
invalid:      0
hdr_len:      224
compression:  l7
load_address: 0x80060000
entry_point:  0x80060000
timestamp:    Tue May 11 08:44:02 2021
binl7_len:    2076978
hdr_version:  4
hdr_cksum:    0x6669 != 0x225C - bad cksum
version:      OPENWRT123456789	( OPENWRT12345678 )
MD5:          2620687EA9670223BACEAAD858496AA4
product:      zf7752	(0)
architecture: 1
chipset:      3
board_type:   0
board_class:  3
customer:
---------------<Image2 FW header>
Magic:        RCKS
next_image:   0x130000
invalid:      0
hdr_len:      160
compression:  l7
load_address: 0x80080000
entry_point:  0x80300A90
timestamp:    Mon Jan 18 04:06:50 2016
binl7_len:    17940320
hdr_version:  4
hdr_cksum:    0x43C0
version:      200.1.9.12.55	( 200.1.9.12.55 )
MD5:          5414A448D69E6ECD070F83C4FDE8B86A
product:      zf7752	(0)
architecture: 1
chipset:      3
board_type:   0
board_class:  3
customer:
OK